gui.py

```diff
@@ -401,11 +404,16 @@ def __get_kontaktdaten(self, modus: Modus) -> dict:
         Returns:
             dict: Kontakdaten
         """
-
+        bDialogAborted = False
         if not os.path.isfile(self.pfad_kontaktdaten):
-            self.kontaktdaten_erstellen(modus)
-
+            bDialogAborted = not self.kontaktdaten_erstellen(modus)
+         
         kontaktdaten = kontak_tools.get_kontaktdaten(self.pfad_kontaktdaten)
+       
+        # we need to check data if we got no dict or the dialog was aborted 
+        if bDialogAborted is True and not kontaktdaten:
+            return {}
+
```

Comment:
Mit dem Bool wert wo du nun zurück bekommst würde ich anders umgehen:
```suggestion
        if not os.path.isfile(self.pfad_kontaktdaten):
            if not self.kontaktdaten_erstellen(modus):
                return {}
```

Reply:
Sowas hatte ich auch bereits gedacht aber du kannst ja im Dialog auch ein File öffnen und dann cancel drücken, dann hat er ja bereits da neue File als referenz und zieht das auch ran für die Suche, ausser man will das nicht dann ist dein Change natürlich deutlich schöner ;)

Reply:
ah ok, ja ich weiß was du meinst
Aber ich glaube das ist bisschen komisch, wenn ich auf "Termin suchen" drücke, dann ne Datei Lade und dann sage "halt abbrechen". Dann sollte man ja auch abbrechen und nicht auf einmal die Suche starten :D

Reply:
Stimmt 😁😁😁😁😁😁😁.
Werde es abändern